crates/ty_completion_eval/src/main.rs

```diff
@@ -0,0 +1,588 @@
+        value_name = "FLOAT",
+        default_value_t = 0.001
+    )]
+    threshold: f64,
+    /// If given, a CSV file of the results for each individual task
+    /// is written to the path given.
+    #[arg(
+        long,
+        help = "When provided, write individual task results in CSV format.",
+        value_name = "FILE"
+    )]
+    tasks: Option<String>,
+}
+
+#[derive(Debug, clap::Parser)]
+struct ShowOneCommand {
+    /// The name of one or more completion tasks to run in isolation.
+    ///
+    /// The name corresponds to the name of a directory in
+    /// `./crates/ty_completion_eval/truth/`.
+    #[arg(help = "The task name to run.", value_name = "TASK_NAME")]
+    task_name: String,
+    /// The name of the file, relative to the root of the
+    /// Python project, that contains one or more completion
+    /// tasks to run in isolation.
+    #[arg(long, help = "The file name to run.", value_name = "FILE_NAME")]
+    file_name: Option<String>,
+    /// The index of the cursor directive within `file_name`
+    /// to select.
+    #[arg(
+        long,
+        help = "The index of the cursor directive to run.",
+        value_name = "INDEX"
+    )]
+    index: Option<usize>,
+}
+
+impl ShowOneCommand {
+    fn matches_source_task(&self, task_source: &TaskSource) -> bool {
+        self.task_name == task_source.name
+    }
+
+    fn matches_task(&self, task: &Task) -> bool {
+        self.task_name == task.name
+            && self
+                .file_name
+                .as_ref()
+                .is_some_and(|name| name == task.cursor_name())
+            && self.index.is_some_and(|index| index == task.cursor.index)
+    }
+}
+
+fn main() -> anyhow::Result<ExitCode> {
+    let args = Cli::parse();
+
+    // The base path to which all CLI arguments are relative to.
+    let cwd = {
+        let cwd = std::env::current_dir().context("Failed to get the current working directory")?;
+        SystemPathBuf::from_path_buf(cwd).map_err(|path| {
+            anyhow!(
+                "The current working directory `{}` contains non-Unicode characters. \
+                 ty only supports Unicode paths.",
+                path.display()
+            )
+        })?
+    };
+    // Where we store our truth data.
+    let truth = cwd.join("crates").join("ty_completion_eval").join("truth");
+    anyhow::ensure!(
+        truth.as_std_path().exists(),
+        "{truth} does not exist: ty's completion evaluation must be run from the root \
+         of the ruff repository",
+        truth = truth.as_std_path().display(),
+    );
+    // The temporary directory at which we copy our truth
+    // data to. We do this because we can't use the truth
+    // data as-is with its `<CURSOR>` annotations (and perhaps
+    // any other future annotations we add).
+    let tmpdir = std::env::var_os("TMPDIR")
+        .and_then(|path| SystemPathBuf::from_path_buf(PathBuf::from(path)).ok())
+        .unwrap_or_else(|| SystemPathBuf::from("/tmp"));
```

Comment:
Could we use the `tempdir` crate here, which works cross platform and cleans up once it's done

Reply:
I did it this way intentionally to avoid clean-up so that `uv sync` doesn't have to do as much work on repeated runs. It's also easier to debug when something goes wrong. But maybe that is a premature optimization.

I've updated this to use `tempfile` and added a flag that lets you opt into keeping the temporary directory around after the eval is done.